client/src/containers/my-projects/index.tsx

```diff
@@ -26,44 +32,63 @@ import {
   TableHeader,
   TableRow,
 } from "@/components/ui/table";
-import TablePagination from "@/components/ui/table-pagination";
+import TablePagination, {
+  PAGINATION_SIZE_OPTIONS,
+} from "@/components/ui/table-pagination";
 import { Tabs, TabsList, TabsTrigger } from "@/components/ui/tabs";
 
 import { columns } from "./columns";
 import Header from "./header";
-import { CustomProject, CustomColumn } from "./types";
 
 const LAYOUT_TRANSITIONS = {
   duration: 0.2,
   ease: "easeInOut",
 };
 
-export default function MyProjectsView({
-  data,
-  filters,
-}: {
-  data: CustomProject[];
-  filters: { label: string; count: number }[];
-}) {
-  const { open: navOpen } = useSidebar();
-
+export default function MyProjectsView() {
+  const { data: session } = useSession();
+  const [pagination, setPagination] = useState<PaginationState>({
+    pageIndex: 0,
+    pageSize: Number.parseInt(PAGINATION_SIZE_OPTIONS[0]),
+  });
   const [sorting, setSorting] = useState<SortingState>([
     {
       id: "projectName",
       desc: false,
     },
   ]);
-  const [pagination, setPagination] = useState<PaginationState>({
-    pageIndex: 0,
-    pageSize: 10,
-  });
-  const [activeFilter, setActiveFilter] = useState("All");
+  const [filters, setFilters] = useMyProjectsFilters();
+  const queryKey = queryKeys.customProjects.all({
+    ...filters,
+    sorting,
+    pagination,
+  }).queryKey;
 
+  const { data } = client.customProjects.getCustomProjects.useQuery(
+    queryKey,
+    {
+      query: {
+        include: ["country"],
+        filter: {
+          activity: filters.activity,
+        },
+        partialProjectName: filters.keyword || undefined,
+        pageNumber: pagination.pageIndex + 1,
+        pageSize: pagination.pageSize,
+      },
+      extraHeaders: {
+        authorization: `Bearer ${session?.accessToken as string}`,
+      },
+    },
+    {
+      select: (d) => d.body,
+      queryKey,
+    },
```

Comment:
@atrincas can you try adding [keepPreviousData](https://github.com/Vizzuality/tnc-blue-carbon-cost-tool/blob/4028cf98ddf4fc2d78d646cd2da09c54d81c8fc0/client/src/containers/overview/table/view/overview/index.tsx#L95)? This way the results don't blink whenever the user is typing.